File: clients/scala/src/main/scala/ai_cup_22/model/Loot.scala
package ai_cup_22.model

import ai_cup_22.util.StreamUtil

/**
 * Loot lying on the ground
 *
 * @param id Unique id
 * @param position Position
 * @param item Item
 */
case class Loot(id: Int, position: ai_cup_22.model.Vec2, item: ai_cup_22.model.Item) {
    /**
     * Write Loot to output stream
     */
    def writeTo(stream: java.io.OutputStream): scala.Unit = {
        StreamUtil.writeInt(stream, id)
        position.writeTo(stream)
        item.writeTo(stream)
    }

    /**
     * Get string representation of Loot
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Loot { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("position: ")
        stringBuilder.append(position)
        stringBuilder.append(", ")
        stringBuilder.append("item: ")
        stringBuilder.append(item)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Loot {
    /**
     * Read Loot from input stream
     */
    def readFrom(stream: java.io.InputStream): Loot = Loot(
        StreamUtil.readInt(stream),
        ai_cup_22.model.Vec2.readFrom(stream),
        ai_cup_22.model.Item.readFrom(stream)
    )
}

File: clients/swift/Sources/AiCup22/Model/Loot.swift
/// Loot lying on the ground
public struct Loot {
    /// Unique id
    let id: Int32

    /// Position
    let position: Vec2

    /// Item
    let item: Item

    /// Read Loot from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Loot {
        var id: Int32
        id = stream.readInt32()
        var position: Vec2
        position = Vec2.readFrom(stream)
        var item: Item
        item = Item.readFrom(stream)
        return Loot(id: id, position: position, item: item)
    }

    /// Write Loot to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(id)
        position.writeTo(stream)
        item.writeTo(stream)
    }
}